editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -1459,14 +1459,15 @@ Transform3D Node3DEditorViewport::_compute_transform(TransformMode p_mode, const
 				p_motion.snap(Vector3(p_extra, p_extra, p_extra));
 			}
 
+			Transform3D t;
 			if (p_local) {
-				p_motion = p_original.basis.xform(p_motion);
+				t = p_original_local;
+				t.origin += p_original_local.basis.xform(p_motion);
+			} else {
+				t = p_original;
+				t.origin += p_motion;
 			}
 
-			// Apply translation
-			Transform3D t = p_original;
-			t.origin += p_motion;
-
 			return t;
```

Comment:
BTW there are dedicated methods for such operations, it could be simplified to e.g.:
```cpp
if (p_local) {
	return p_original_local.translated_local(p_motion);
} else {
	return p_original.translated(p_motion);
}

Reply:
Thank you! Will test that